### PR TITLE
BPE pretokenizer - add support for command-r-plus and command-r models

### LINKED ISSUE #7030: Command-R GGUF conversion no longer working
As recently as a few days ago, Command-R (and presumably R+) could be converted with convert-hf-to-gguf.py.  I double checked and conversion completes successfully in b2751.  However, with the recent changes to accommodate Llama3, Command-R compatibility has been broken.  Trying to convert today with b2777 I get

```
raise NotImplementedError("BPE pre-tokenizer was not recognized - update get_vocab_base_pre()")
NotImplementedError: BPE pre-tokenizer was not recognized - update get_vocab_base_pre()
```

I know that L3 required a new tokenizer provided by meta to facilitate proper conversion.  Do we require something new from cohere, or is this something that can be fixed internally?

### LINKED ISSUE #7040: Command-R-Plus unable to convert or use after BPE pretokenizer update
Trying to re-convert the CohereForAI/command-r-plus model after BPE updates. Model not yet added.

Use `python convert-hf-to-gguf.py`

If the bug concerns the server, please try to reproduce it first using the [server test scenario framework](https://github.com/ggerganov/llama.cpp/tree/master/examples/server/tests).

### PR DESCRIPTION
**Pull request by sealad886**
Fixes #7030 
Fixes #7040 

Added both command-r and command-r-plus pretokenizer regex's to the model, updated the files as recommended in ([120cf37](https://github.com/ggerganov/llama.cpp/pull/6920/commits/120cf37d546a82042bd1ad5894e4814a0efa8086)). 

Tests compiled at `./tests/test-tokenizer-o` passed for both models.

### CODE REVIEW

**Comment by dranger003** [comment]
There is already a PR for this https://github.com/ggerganov/llama.cpp/pull/7033.

File: convert-hf-to-gguf.py [hunk 1]
```diff
@@ -306,6 +306,12 @@ def get_vocab_base_pre(self, tokenizer) -> str:
         if chkhsh == "3ce83efda5659b07b1ad37ca97ca5797ea4285d9b9ab0dc679e4a720c9da7454":
             # ref: https://huggingface.co/openai-community/gpt2
             res = "gpt-2"
+        if chkhsh == "9c2227e4dd922002fb81bde4fc02b0483ca4f12911410dee2255e4987644e3f8":
+            # ref: https://huggingface.co/CohereForAI/c4ai-command-r
+            res = "command-r"
+        if chkhsh == "9c2227e4dd922002fb81bde4fc02b0483ca4f12911410dee2255e4987644e3f8":
```
Comment (JoanFM) [inline convert-hf-to-gguf.py:312]: just curious, aren't they the exact same hash?

**Comment by LostRuins** [comment]
Is this duplicated in https://github.com/ggerganov/llama.cpp/pull/7063 ?

**Comment by dranger003** [comment]
@LostRuins Yes, this is a duplicate.

**Pull request closed without merging**